### ISSUE TITLE
Bracket-notation for property access is not working

### ISSUE THREAD

**Issue by jasonpacheco** [TroyAlford/react-jsx-parser#125]
Hi Troy,

My team is building a functionality where end-users can edit JSX templates and integrate their changes into the application. Your parser works very well for most of the features we're trying to implement, but where we're having difficulty is supporting bracket notation inside our JSX templates and SVG markup.

In the below snippet, `data[0].name` will not parse because of bracket notation, but if I do something like `data.zero.name`, it parses correctly. My workaround for this is passing a function into `bindings` like `getValue` that can get the desired property using bracket notation, but it's written as a string. This is not a user-friendly solution, and we ideally want users to write JSX without resorting to these workarounds. 

```javascript
import get from 'lodash.get';

const template = '<div><p>{data[0].name}</p></div>';
const workaround = "<div><p>{getValue.call(this, data, '[0].name'}</p></div>";
...
<JsxParser 
  jsx={template} 
  bindings={{
    data: {
      0: {
        name: 'user'
      }
    },
    getValue: (obj, prop) => get(obj, prop)
  }} 
/>
```
Same thing with SVG markup. We would like users to add their own SVG markup inline with the JSX templates but the parser can't parse inline SVG unless it is passed as a unique component into `component`.

We are using the latest 1.23.0 version. Unless I'm missing the obvious, do you have any suggestions for these limitations we're having and if we fork the project, could you point us in the right direction to implement our desired functionality?

Thank you!

**Comment by TroyAlford** [comment]
It looks like we aren't correctly indexing into numeric properties. Your example doesn't work, currently, with `data: [{ name: 'user' }]` either, or with `data["0"].name`. We really should be allowing any valid `["string"]` syntax as a data accessor, or we limit things like: `data: { "first name": "john" }` as well (which can only be indexed properly as `data["first name"].name`

What's happening, currently, I think, is that the parser gets to this property name and doesn't find it - so it returns `null`. As a result, you get a blank value in your div.

As for SVG, I'm going to open a separate ticket for you, momentarily.

**Comment by TroyAlford** [comment]
This is now released as [v1.24.0](https://github.com/TroyAlford/react-jsx-parser/releases/tag/v1.24.0)

**Comment by jasonpacheco** [comment]
Thank you for addressing this, Troy! I don't mean to be the bearer of bad news, but I think the latest release broke the usage of methods like in #105 when using them directly with nested objects. In v1.23.0, `{name.first.toUpperCase()}` parsed correctly as`NAME`, but it parses as`[OBJECT OBJECT]` in v1.24.0.

 Another example: `{object.someNumber.toFixed(2)}` causes nothing to render and gives me the error: `Number.prototype.toFixed requires that 'this' be a Number`.

Here is a CodeSandbox: https://codesandbox.io/s/jsx-parser-issues-15pn9?file=/src/App.js

**Comment by TroyAlford** [comment]
Woops - yep, I think I know exactly what I did wrong, here. Thanks for the quick report. I should've added a unit test for a deeply nested function accessor like this.

**Comment by TroyAlford** [comment]
I had changed it so that if you bound `foo.bar.baz(5)` it would essentially result in `baz.call(foo, 5)` instead of what it should be: `baz.call(baz, 5)`. I have a fix and a new unit test - and will release this in a few min.

**Comment by TroyAlford** [comment]
@jasonpacheco - this is now patched and released as [v1.24.1](https://github.com/TroyAlford/react-jsx-parser/releases/tag/v1.24.1). If you see anything else, please feel free to open a new ticket!

**Comment by jasonpacheco** [comment]
@TroyAlford Thank you for the quick fix!
